aten/src/THC/generic/THCTensorMath.cu

```diff
@@ -139,6 +139,11 @@ void THCTensor_(catArray)(THCState *state, THCTensor *result,
     }
   }
 
+  // If all inputs are empty tensors, return an empty tensor
+  if (notEmptyTensor == NULL) {
+    return;
```

Comment:
Is the output tensor in TH is initialized as empty tensor?

Reply:
Yes. And THC.

Reply:
In `CPUFloatType::_cat`, the tensor is initialized using `THFloatTensor_new`. `THTensor_(new)` calls the following function in THTensor.cpp
```
static void THTensor_(rawInit)(THTensor *self)
{
  self->refcount = 1;
  self->storage = THStorage_(new)();
  self->storageOffset = 0;
  self->size = NULL;
  self->stride = NULL;
  self->nDimension = 0;
  self->flag = TH_TENSOR_REFCOUNTED;
}
```
which initializes an empty tensor. I expect the same to happen for the THC code.